### PR TITLE
CORDA-3068: Pass base directory when resolving relative paths

### PR DESCRIPTION
**Pull request by vkolomeyko**
We have 2 different methods/frameworks:

1. Reflection based ConfigUtilities from NodeAPI. This has no effect on Node configuration parsing.
2. Specification driven parsing which is used for Node configuration written by @sollecitom

### CODE REVIEW

**Comment by shamsasari** [comment]
This doesn't actually change how the node config file is parsed, correct?

**Comment by vkolomeyko** [comment]
> This doesn't actually change how the node config file is parsed, correct?

@shamsasari - there is a small change to `V1NodeConfigurationSpec.kt` which slightly changes the way how node config parsed. I have all added a unit test to make sure the change has desired effect.

**Comment by vkolomeyko** [comment]
@anthonykeenan @shamsasari and @mnesbit - could you please review this PR and let me if you are comfortable with changes? Thanks!

File: node/src/main/kotlin/net/corda/node/services/config/schema/v1/V1NodeConfigurationSpec.kt [hunk 1]
```diff
@@ -80,10 +80,11 @@ internal object V1NodeConfigurationSpec : Configuration.Specification<NodeConfig
 
         val messagingServerExternal = configuration[messagingServerExternal] ?: Defaults.messagingServerExternal(configuration[messagingServerAddress])
         val database = configuration[database] ?: Defaults.database(configuration[devMode])
-        val cordappDirectories = configuration[cordappDirectories] ?: Defaults.cordappsDirectories(configuration[baseDirectory])
+        val baseDirectoryPath = configuration[baseDirectory]
+        val cordappDirectories = configuration[cordappDirectories] ?: Defaults.cordappsDirectories(baseDirectoryPath)
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/config/schema/v1/V1NodeConfigurationSpec.kt]: Is there any reason why you've changed the behaviour of `cordappDirectories`, as it's not actually a documented config option? It was added as part of some change to the driver dsl code, I personally wish it hadn't been introduced - if anything it should have been a simple `cordappDirectory` option.

**Comment by shamsasari** [comment]
You've changed `cordappDirectories`, which isn't actually a documented config option. Has the behaviour of the node config with respect to the other directory configs also changed? If it has then it needs to be documented as a operational behaviour change. Judging from the code change I don't actually think it has changed, but I can't be 100% sure.

**Comment by vkolomeyko** [comment]
@shamsasari 
> Has the behaviour of the node config with respect to the other directory configs also changed?
Nope - no changes to other directory configs. And, actually, there are no other directory configurations that are already not dependant on `baseDirectory` in Corda OS.

File: node/src/main/kotlin/net/corda/node/services/config/schema/v1/V1NodeConfigurationSpec.kt (hunk 1 above)
Comment (vkolomeyko) [reply node/src/main/kotlin/net/corda/node/services/config/schema/v1/V1NodeConfigurationSpec.kt]: Having realised that changes to `node-api\src\main\kotlin\net\corda\nodeapi\internal\config\ConfigUtilities.kt` have no effect on parsing of the NodeConfiguration in order to change behaviour described in https://r3-cev.atlassian.net/browse/CORDA-3068 I made changes to `V1NodeConfigurationSpec.kt`

But given that:
a) `cordappDirectories` is not a documented option;
and
b) Change is trivial, arguably good and supported by the test.
Are there any particular concerns for having it?

**Pull request merged**